### ISSUE TITLE
Add a documentation section about containers/kubernetes

### ISSUE THREAD

**Issue by grgrzybek** [jolokia/jolokia#618]
This chapter could include:
* run-java.sh
* patterns for adding Jolokia agent (`-javaagent`) to container-based application
* ...

cc: @rhuss @tadayosi @apupier - I won't do it for 2.0.0 GA, but we need it later.

**Comment by tadayosi** [comment]
Sounds great.

Is run-java.sh still relevant?  Just unlike AB Jolokia?

**Comment by grgrzybek** [comment]
I found it (among others):
* https://github.com/containers-tools/os-java/blob/master/os-java-run/added/run-java.sh
* https://github.com/jboss-openshift/cct_module/blob/75645e2/os-java-run/added/run-java.sh
* https://codergists.com/redhat/containers/java/openshift/2020/01/16/what-makes-up-a-red-hat-java-based-container.html

but even if, it'd be good to mention the ways to transparently (or in controlled way using ENV variables in docker/podman invocations) add Jolokia to your containerized Java application.

**Comment by rhuss** [comment]
`run-java.sh`'s main purpose was to set the memory and cpu constraints for older JVM version (early Java 8 versions), where the defaults were calculated based on the host, not the container. It also adds Jolokia to the mix (and "agent-bond", a super agent that combines Jolokia and jmx_exporter) which probably was a bad idea and later removed). We could refactor this to remove the unneeded parts, but we should still evaluate whether having and maintaining a startup script is a good idea. However, this is probably a topic for different repo's issue :-)

**Comment by rhuss** [comment]
I agree that we should add some recommendations how to add Jolokia in a Kubernetes context.

**Comment by tadayosi** [comment]
Worth noting that JKube prebuilds Jolokia in its default base image. It should be our recommended way to deploy Jolokia in a K8s cluster.
https://eclipse.dev/jkube/docs/kubernetes-maven-plugin/#zero-config

We should not exclude other options from OSS community (if there are available), of course.
